nomad/state/state_store_test.go

```diff
@@ -6587,6 +6587,30 @@ func TestStateStore_UpsertAlloc_ChildJob(t *testing.T) {
 	require.False(t, watchFired(ws))
 }
 
+func TestStateStore_UpsertAlloc_NextAllocation(t *testing.T) {
+	ci.Parallel(t)
+
+	state := testStateStore(t)
+
+	alloc1 := mock.Alloc()
+	alloc2 := mock.Alloc()
+	alloc2.PreviousAllocation = alloc1.ID
+
+	err := state.UpsertAllocs(structs.MsgTypeTestSetup, 1000, []*structs.Allocation{alloc1, alloc2})
+	must.NoError(t, err)
+
+	// alloc1 should have the correct NextAllocation
+	actual, err := state.AllocByID(nil, alloc1.ID)
+	must.Eq(t, actual.NextAllocation, alloc2.ID)
+
+	err = state.UpsertAllocs(structs.MsgTypeTestSetup, 1001, []*structs.Allocation{alloc2, alloc1})
+	must.NoError(t, err)
+
+	// upsert in a different order, alloc1 should still have the correct NextAllocation
+	actual, err = state.AllocByID(nil, alloc1.ID)
```

Comment:
This will never return an error unless you give it a non-UUID ID, but we should assert that here anyways.
```suggestion
	actual, err = state.AllocByID(nil, alloc1.ID)
	must.NoError(t, err)
```

Reply:
Good call, thanks!